Okta.Xamarin/Okta.Xamarin/IOidcClient.cs

```diff
@@ -27,19 +29,60 @@ public interface IOidcClient
         /// </summary>
         /// <param name="sessionToken">A valid session  token obtained via the <see cref="https://github.com/okta/okta-auth-dotnet">AuthN SDK</see></param>
         /// <returns>In case of successful authorization, this Task will return a valid <see cref="OktaStateManager"/>.  Clients are responsible for further storage and maintenance of the manager.</returns>
-        Task<OktaStateManager> AuthenticateAsync(string sessionToken);
+        Task<IOktaStateManager> AuthenticateAsync(string sessionToken);
 
         /// <summary>
         /// Start the authorization flow.  This is an async method and should be awaited.
         /// </summary>
         /// <returns>In case of successful authorization, this Task will return a valid <see cref="OktaStateManager"/>.  Clients are responsible for further storage and maintenance of the manager.</returns>
-        Task<OktaStateManager> SignInWithBrowserAsync();
+        Task<IOktaStateManager> SignInWithBrowserAsync();
 
         /// <summary>
         /// This method will end the user's Okta session in the browser.  This is an async method and should be awaited.
         /// </summary>
         /// <param name="stateManager">The state manager associated with the login that you wish to log out</param>
         /// <returns>Task which tracks the progress of the logout</returns>
-        Task<OktaStateManager> SignOutOfOktaAsync(OktaStateManager stateManager);
-	}
-}
\ No newline at end of file
+        Task<IOktaStateManager> SignOutOfOktaAsync(IOktaStateManager stateManager);
+
+        /// <summary>
+        /// Revokes the specified access token.
+        /// </summary>
+        /// <param name="accessToken">The access token.</param>
+        /// <returns>Task.</returns>
+        Task RevokeAccessTokenAsync(string accessToken);
+
+        /// <summary>
+        /// Revokes the specified refresh token.
+        /// </summary>
+        /// <param name="accessToken">The access token used for authorization.</param>
+        /// <param name="refreshToken">The refresh token.</param>
+        /// <returns>Task.</returns>
+        Task RevokeRefreshTokenAsync(string accessToken, string refreshToken);
+
+        /// <summary>
+        /// Gets user information.
+        /// </summary>
+        /// <param name="accessToken">The access token used for authorization.</param>
+        /// <param name="authorizationServerId">The authorization server id or null.</param>
+        /// <returns>Task{Dictionary{string, object}}</returns>
+        Task<Dictionary<string, object>> GetUserAsync(string accessToken, string authorizationServerId = "default");
```

Comment:
The Authorization server ID should be part of the configuration. We should get that info by ourselves instead of requesting the info again.  Thoughts?

Reply:
I can add it to the config; I honestly don't know the use case for when it should be specified and when `default` is okay.  I can add a follow up ticket to make this a configuration setting, but it seems like it should be the end developers discretion whether to put it in the config or get it from somewhere else.

Reply:
Added OKTA-373330 for follow up.